### PR TITLE
fix: make sure `head` can be used in `DeepPot`

### PR DESCRIPTION
**Pull request by wangzyphysics**
Following the example mentioned [here](https://www.aissquare.com/models/detail?pageType=models&name=DPA-2.3.0-v3.0.0b4&id=279) , I first select a `head` and put it into `DPCalculator` but encounter the error like this:
```shell
(/public/home/mzq001/soft/deepmd-kit) [mzq001@login01 ~]$ python test.py 
To get the best performance, it is recommended to adjust the number of threads by setting the environment variables OMP_NUM_THREADS, DP_INTRA_OP_PARALLELISM_THREADS, and DP_INTER_OP_PARALLELISM_THREADS. See https://deepmd.rtfd.io/parallelism/ for more information.
Traceback (most recent call last):
  File "/public/home/mzq001/test.py", line 4, in <module>
    dp = DPCalculator("DPA2_medium_28_10M_beta4.pt", head="H2O_H2O-PD")
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/public/home/mzq001/soft/deepmd-kit/lib/python3.11/site-packages/deepmd/calculator.py", line 92, in __init__
    self.dp = DeepPot(str(Path(model).resolve()), neighbor_list=neighbor_list)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/public/home/mzq001/soft/deepmd-kit/lib/python3.11/site-packages/deepmd/infer/deep_eval.py", line 334, in __init__
    self.deep_eval = DeepEvalBackend(
                     ^^^^^^^^^^^^^^^^
  File "/public/home/mzq001/soft/deepmd-kit/lib/python3.11/site-packages/deepmd/pt/infer/deep_eval.py", line 121, in __init__
    head is not None
AssertionError: Head must be set for multitask model! Available heads are: ['Domains_Alloy', 'Domains_Anode', 'Domains_Cluster', 'Domains_Drug', 'Domains_FerroEle', 'Domains_OC2M', 'Domains_SSE-PBE', 'Domains_SemiCond', 'H2O_H2O-PD', 'Metals_AgAu-PBE', 'Metals_AlMgCu', 'Metals_Cu', 'Metals_Sn', 'Metals_Ti', 'Metals_V', 'Metals_W', 'Others_C12H26', 'Others_HfO2', 'Domains_ANI', 'Domains_SSE-PBESol', 'Domains_Transition1x', 'H2O_H2O-DPLR', 'H2O_H2O-PBE0TS-MD', 'H2O_H2O-PBE0TS', 'H2O_H2O-SCAN0', 'Metals_AgAu-PBED3', 'Others_In2Se3', 'MP_traj_v024_alldata_mixu']
```

```python
## Compute potential energy
from ase import Atoms
from deepmd.calculator import DP as DPCalculator
dp = DPCalculator("DPA2_medium_28_10M_beta4.pt", head="H2O_H2O-PD")
water = Atoms('H2O', positions=[(0.7601, 1.9270, 1), (1.9575, 1, 1), (1., 1., 1.)], cell=[100, 100, 100])
water.calc = dp
print(water.get_potential_energy())
print(water.get_forces())

## Run BFGS structure optimization
from ase.optimize import BFGS
dyn = BFGS(water)
dyn.run(fmax=1e-6)
print(water.get_positions())
```

The variable named `head` of `DPCalculator` is not being used in `DeepPot` which directly leads to this error.

## Summary by CodeRabbit

- **New Features**
	- Enhanced configurability of the DP class with an additional `head` parameter for initialization.
	- Updated default behavior in the calculate method to include stress calculations alongside energy and forces.

- **Bug Fixes**
	- Clarified handling of stress property in the context of lattice relaxation, improving accuracy in calculations.

### CODE REVIEW

File: deepmd/calculator.py [hunk 1]
```diff
@@ -87,7 +87,9 @@ def __init__(
         **kwargs,
     ) -> None:
         Calculator.__init__(self, label=label, **kwargs)
-        self.dp = DeepPot(str(Path(model).resolve()), neighbor_list=neighbor_list)
+        self.dp = DeepPot(
+            str(Path(model).resolve()), neighbor_list=neighbor_list, **kwargs
+        )
```
Comment (njzjz) [inline deepmd/calculator.py:92]: I am wondering if two `**kwargs` conflict. Besides, we don't have a doc for `**kwargs`.

Comment (wangzyphysics) [reply deepmd/calculator.py:92]: Extracting the `head` variable might improve clarity.

**Comment by iProzd** [comment]
@wangzyphysics By the way, I apologize for any inconvenience you experienced with the example. I have updated the [example](https://www.aissquare.com/models/detail?pageType=models&name=DPA-2.3.0-v3.0.0b4&id=279), and you can use an external freeze method as a temporary workaround. Thank you for identifying this issue. Regarding the interface itself, we indeed need to add this variable. Let's address and fix it within this PR.

File: deepmd/calculator.py (hunk 1 above)
Comment (iProzd) [reply deepmd/calculator.py:92]: Maybe we can make `head` an explicit argument as `neighbor_list` here.

File: deepmd/calculator.py [hunk 2]
```diff
@@ -84,10 +84,15 @@ def __init__(
         label: str = "DP",
         type_dict: Optional[dict[str, int]] = None,
         neighbor_list=None,
+        head=None,
```
Comment (njzjz) [inline deepmd/calculator.py:87]: Could you add the new argument into the doc, i.e. after line 47?

Comment (wangzyphysics) [reply deepmd/calculator.py:89]: already done!

**Review by njzjz — APPROVED** [review]

**Review by iProzd — APPROVED** [review]

**Pull request merged**